### PR TITLE
Add Babel (JavaScript transpiler) packages to nodePackages

### PR DESCRIPTION
**Pull request by 8573**
These patches add to `nodePackages` many packages from NPM for the
JavaScript "transpiler" Babel, specifically —
- `babel-cli`,
- `babel-eslint`,
- `babel-polyfill`,
- Babel's "presets", and
- Babel's many "transforms".

Additionally, the regeneration of the generated `nodePackages` files
resulted in the `pbkdf2` package being updated.

I have tested the added Babel packages per nixpkgs manual section 10.1
("Making patches"), on a NixOS system on the `nixos-unstable` channel
with build sandboxing enabled.

### CODE REVIEW

**Comment by svanderburg** [comment]
Some of the packages are OK for inclusion (e.g. babel-cli), but for others I'm not entirely sure whether they should be in Nixpkgs.

In principle, we should only include *end-user* NPM packages in Nixpkgs (such as command-line tools), but not any libraries or plugins, unless they are used by non-NPM packages in Nixpkgs.

The biggest reason why we should exclude libraries/plugins is that NPM dependencies compose differently depending on their contexts. This is something npm2nix did not simulate very well. In contrast, node2nix does dependency management differently (and more accurately) and makes the inclusion of libraries more expensive. 

The second reason why not all packages can be included, is because node2nix composes entire dependency trees in one derivation, so it makes no sense to refer to them the "Nix way".

If you want to know more about NPM dependency "simulation" in Nix: I have written an elaborate blog post about this a while ago: http://sandervanderburg.blogspot.com/2016/02/managing-npm-flat-module-installations.html

Typically, babel plugins are supposed to be installed as devDependencies of a development project. A better workflow would be to directly use node2nix on a package.json file of a development project.

Consult the README.md file of node2nix for more information on how to use node2nix in development projects. I think section: 'Adding additional/global NPM packages to a packaging process' is particularly important -- it describes how to use grunt and grunt plugins. I think the same workflow applies to babel (and its corresponding plugins).

I also think that using node2nix on a package.json file is much more convenient than manually composing Nix expressions that refer to NPM library packages in Nixpkgs.

So to summarize: if you intend to use Nix for a Node.js development project, use node2nix directly on the project to provide it the libraries it needs.

**Comment by 8573** [comment]
Ah. My apologies for filing this PR without understanding nixpkgs' NPM packaging.

**Comment by 8573** [comment]
Are linters like `htmlhint` appropriate for inclusion?

**Comment by svanderburg** [comment]
@8573 Yes that is a command-line utility
